src/corona_hakab_model/generation/geographic_circle.py

```diff
@@ -34,22 +36,34 @@ def generate_agents_ages_and_connections_types(self):
         Iterates over each agent. Generates it's age by a given age distribution (self.data_holder.age_distribution).
         For each agent, iterates over each connection type, and rolls whether it has this type of connection or not.
         Gets this probabilities from self.data_holder.connection_types_prob_by_age.
-        It allows choosing whether some one goes to work, to school, or to none of those.
+        It allows choosing whether some one goes to work, to school, to kindergarten or to none of those.
         :return:
         """
         ages = iter(self.data_holder.age_distribution.rvs(size=len(self.agents)))
+        workplace_distribution = rv_discrete(
+            values=([ConnectionTypes.School, ConnectionTypes.Kindergarten, ConnectionTypes.Work],
+                    [self.data_holder.teachers_workforce_ratio, self.data_holder.kindergarten_workforce_ratio,
+                    1 - self.data_holder.teachers_workforce_ratio - self.data_holder.kindergarten_workforce_ratio])
+        )
         for agent, age in zip(self.agents, ages):
             agent.age = age
             agent_connection_types = []
-            for connection_type in ConnectionTypes:
+            for connection_type in Non_Exclusive_Types:
```

Comment:
I an not sure that Non_Exclusive_Types is very clear. rather it's better to explicitly handle Work/School/Kindergarden, and all connection types not in [Work/School/Kindergarden].

Reply:
In my opinion, the NonExclusive_type is slightly more readable